util/src/main/java/tech/pegasys/artemis/util/backing/ViewType.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.util.backing;
+
+import tech.pegasys.artemis.util.backing.tree.TreeNode;
+
+/**
+ * Base class for any SSZ type like Vector, List, Container, basic types
+ * (https://github.com/ethereum/eth2.0-specs/blob/dev/ssz/simple-serialize.md#typing)
+ */
+public interface ViewType {
+
+  /**
+   * Creates a default backing binary tree for this type E.g. if the type is basic then normally
+   * just a single leaf node is created E.g. if the type is a complex structure with multi-level
+   * nested vectors and containers then the complete tree including all descendant members subtrees
+   * is created
+   */
+  TreeNode getDefaultTree();
+
+  /**
+   * Creates immutable View over the tree which should correspond to this type If the tree structure
+   * doesn't correspond this type that fact could only be detected later during access to View
+   * members
+   */
+  ViewRead createFromTreeNode(TreeNode node);
+
+  /** Creates a default immutable View */
+  default ViewRead getDefault() {
+    return createFromTreeNode(getDefaultTree());
+  }
+
+  /**
+   * Returns the number of bits the element of this type occupies in a tree node More correct
+   * definition is: how many elements of this type one tree node may contain All complex types
+   * occupies the whole tree node and their bitsize assumed to be 256 Normally the bitsize < 256 is
+   * for basic types that can be packed into a single leaf node
+   */
+  default int getBitsSize() {
```

Comment:
In general, I'd avoid setting defaults that only apply to some subtypes.  
Suggest moving this default implementation to `CompositeViewType`:
```suggestion
  int getBitsSize();
```

Reply:
Yep, very good general rule 👍 
Changed.